src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1583,448 +1719,66 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
-                embOp2Node->GetResultOpNumForRmwIntrinsic(user, intrinEmb.op1, intrinEmb.op2, intrinEmb.op3);
-
-            if (resultOpNum == 0)
-            {
-                prefUseNode = embOp2Node->Op(1);
-            }
-            else
-            {
-                assert(resultOpNum >= 1 && resultOpNum <= 3);
-                prefUseNode = embOp2Node->Op(resultOpNum);
-            }
-        }
-        else
-        {
-            const bool embHasImmediateOperand = HWIntrinsicInfo::HasImmediateOperand(intrinEmb.id);
-            assert((numArgs == 1) || (numArgs == 2) || (numArgs == 3) || (embHasImmediateOperand && (numArgs == 4)));
-
-            // Special handling for embedded intrinsics with immediates:
-            // We might need an additional register to hold branch targets into the switch table
-            // that encodes the immediate
-            bool needsInternalRegister;
-            switch (intrinEmb.id)
-            {
-                case NI_Sve_ShiftRightArithmeticForDivide:
-                    assert(embHasImmediateOperand);
-                    assert(numArgs == 2);
-                    needsInternalRegister = !embOp2Node->Op(2)->isContainedIntOrIImmed();
-                    break;
-
-                case NI_Sve_AddRotateComplex:
-                    assert(embHasImmediateOperand);
-                    assert(numArgs == 3);
-                    needsInternalRegister = !embOp2Node->Op(3)->isContainedIntOrIImmed();
-                    break;
+                embeddedOpNode->GetResultOpNumForRmwIntrinsic(user, intrinEmbedded.op1, intrinEmbedded.op2,
+                                                              intrinEmbedded.op3);
 
-                case NI_Sve_MultiplyAddRotateComplex:
-                    assert(embHasImmediateOperand);
-                    assert(numArgs == 4);
-                    needsInternalRegister = !embOp2Node->Op(4)->isContainedIntOrIImmed();
-                    break;
-
-                default:
-                    assert(!embHasImmediateOperand);
-                    needsInternalRegister = false;
-                    break;
-            }
-
-            if (needsInternalRegister)
+            if (resultOpNum != 0)
             {
-                buildInternalIntRegisterDefForNode(embOp2Node);
+                embeddedDelayFreeOp = embeddedOpNode->Op(resultOpNum);
             }
-
-            size_t prefUseOpNum = 1;
-            if (intrinEmb.id == NI_Sve_CreateBreakPropagateMask)
-            {
-                prefUseOpNum = 2;
-            }
-            prefUseNode = embOp2Node->Op(prefUseOpNum);
         }
 
-        for (size_t argNum = 1; argNum <= numArgs; argNum++)
+        for (size_t argNum = 1; argNum <= intrinEmbedded.numOperands; argNum++)
         {
-            GenTree* node = embOp2Node->Op(argNum);
+            GenTree* node = embeddedOpNode->Op(argNum);
 
-            if (node == prefUseNode)
+            if (node == embeddedDelayFreeOp)
```

Comment:
can you add a `TODO-CQ` in `else` part that this should be revisited and improved?